adapter: add session uuid to mz_sessions (#27967)
MaterializeInc/materialize: doc/user/content/sql/system-catalog/mz_internal.md

| `prepared_statement_id`    | [`uuid`]                     | An ID that is unique for each prepared statement. For example, if a statement is prepared once and then executed multiple times, all executions will have the same value for this column (but different values for `execution_id`).                                           |
| `sql_hash`                 | [`bytea`]                    | An opaque value uniquely identifying the text of the query.                                                                                                                                                                                                                   |
| `prepared_statement_name`  | [`text`]                     | The name given by the client library to the prepared statement.                                                                                                                                                                                                               |

> | `session_id`               | [`uuid`]                     | An ID that is unique for each session. Corresponds to [mz_sessions.id](https://materialize.com/docs/sql/system-catalog/mz_catalog/#mz_sessions). |

Reviewer: ```suggestion
| `session_id`               | [`uuid`]                     | An ID that is unique for each session. Corresponds to [`mz_sessions.id`](#mz_sessions). |
```

| Field                | Type                         | Meaning                                                                                                                           |
|----------------------|------------------------------|-----------------------------------------------------------------------------------------------------------------------------------|

> | `id`                 | [`uuid`]                     | The globally unique ID of this history entry. Corresponds to [`mz_sessions.id`](#mz_sessions). |

Reviewer: ```suggestion
| `id`                 | [`uuid`]                     | The globally unique ID of the session. Corresponds to [`mz_sessions.id`](#mz_sessions). |
```

<!-- RELATION_SPEC mz_internal.mz_sessions -->
| Field           | Type                           | Meaning                                                                                                                   |
| --------------- | ------------------------------ | --------                                                                                                                  |

> | `id`            | [`uuid`]                       | The UUID of the session. Globally unique for all sessions. Corresponds to [`mz_recent_activity_log.session_id](#mz_recent_activity_log). |

Reviewer: ```suggestion
| `id`            | [`uuid`]                       | The globally unique ID of the session. |
```

| `id`            | [`uint4`]                      | The ID of the session.                                                                                                    |
| `role_id`       | [`text`]                       | The role ID of the role that the session is logged in as. Corresponds to [`mz_catalog.mz_roles`](../mz_catalog#mz_roles). |
| `connected_at`  | [`timestamp with time zone`]   | The time at which the session connected to the system.                                                                    |

> | `uuid`          | [`uuid`]                       | The UUID of the session. Corresponds to [`mz_recent_activity_log.session_id](#mz_recent_activity_log).                    |

Reviewer 2: Having an ID and a UUID is confusing! At the very least we should explain with more words what the difference is. Possibly we should also pick more descriptive names for these fields. E.g., perhaps `id` should be called `connection_id`, and `uuid` should be called `id`.

Reviewer 3: +1
